x/savings/keeper/deposit.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/savings/types"
+)
+
+// Deposit deposit
+func (k Keeper) Deposit(ctx sdk.Context, depositor sdk.AccAddress, coins sdk.Coins) error {
+
+	err := k.ValidateDeposit(ctx, coins)
+	if err != nil {
+		return err
+	}
+
+	err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, depositor, types.ModuleAccountName, coins)
+	if err != nil {
+		return err
+	}
+
+	currDeposit, foundDeposit := k.GetDeposit(ctx, depositor)
+	amount := coins
+	if foundDeposit {
+		amount = amount.Add(currDeposit.Amount...)
+	}
+	deposit := types.NewDeposit(depositor, amount)
+	k.SetDeposit(ctx, deposit)
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeSavingsDeposit,
+			sdk.NewAttribute(sdk.AttributeKeyAmount, coins.String()),
+			sdk.NewAttribute(types.AttributeKeyDepositor, deposit.Depositor.String()),
+		),
+	)
+
+	return nil
+}
+
+// ValidateDeposit validates a deposit
+func (k Keeper) ValidateDeposit(ctx sdk.Context, coins sdk.Coins) error {
+	for _, coin := range coins {
+		supported := k.IsDenomSupported(ctx, coin.Denom)
+		if supported == false {
+			return types.ErrInvalidDepositDenom
```

Comment:
I would wrap this error with the denom that caused it - currently the error is just

`raw_log: 'failed to execute message; message index: 0: invalid deposit denom'`

Would like:

`raw_log: 'failed to execute message; message index: 0: invalid deposit denom: bnb'`

Reply:
Added in https://github.com/Kava-Labs/kava/pull/1192/commits/b2cbe147ae733740910887e81c5090ece674c7f6